### PR TITLE
Adding network usage to resource calculator

### PR DESCRIPTION
**Pull request by maayarosama**
### Description

Added a bandwidth field in pricing calculator 
### Changes

- Added an nu variable in calculator model
- Added  a calcNU method

### Related Issues

- #2215

### Checklist

- [ ] Tests included
- [x] Build pass
- [x] Documentation
- [ ] Code format and docstrings
- [ ] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

**Review by AlaaElattar — APPROVED** [review]

File: packages/grid_client/src/modules/calculator.ts [hunk 1]
```diff
@@ -49,6 +49,13 @@ class Calculator {
   async calSU(options: SUModel) {
     return options.hru / 1200 + options.sru / 200;
   }
+
+  @expose
+  @validateInput
+  async calNU(options: NUModel) {
+    // what is this number
```
Comment (zaelgohary) [inline packages/grid_client/src/modules/calculator.ts:56]: Please, remove this comment.

File: packages/playground/src/utils/pricing_calculator.ts [hunk 2]
```diff
@@ -19,6 +19,13 @@ export const cruRules = _applyRules([
   max("Maximum allowed cpu cores is 256.", 256),
 ]);
 
+export const nuRules = _applyRules([
+  required("Bandwidth is required."),
+  isInt("Bandwidth must be a valid integer."),
+  min("Minimum allowed bandwidth is 0", 0),
```
Comment (zaelgohary) [inline packages/playground/src/utils/pricing_calculator.ts:25]:
```suggestion
  min("Minimum allowed bandwidth is 0.", 0),
```

**Review by zaelgohary — COMMENTED** [review]
The UI seems a bit off. Can we replace the switches next to the Bandwidth field?

image

**Review by zaelgohary — CHANGES_REQUESTED** [review]
Since network depends on public IP, I think the bandwidth field should be disabled by default and enabled only if the public IP switch is on. Also, the filed tooltip should explain this to the user.

File: packages/grid_client/src/modules/calculator.ts [hunk 3]
```diff
@@ -49,6 +49,13 @@ class Calculator {
...
+    return (options.nu * 1000) / 1e7;
```
Comment (zaelgohary) [inline packages/grid_client/src/modules/calculator.ts:57]:
```suggestion
    return options.nu / 1e4;
```

Comment (xmonader) [reply packages/grid_client/src/modules/calculator.ts:57]: @zaelgohary no, keep it as * 1000 / 1e7 (this explains the flow)

**Comment by xmonader** [comment]
Please move the balance field next to the bandwidth field

**Pull request merged**